pandas/core/frame.py

```diff
@@ -4675,7 +4675,11 @@ def _reindex_multi(
         new_index, row_indexer = self.index.reindex(axes["index"])
         new_columns, col_indexer = self.columns.reindex(axes["columns"])
 
-        if row_indexer is not None and col_indexer is not None:
+        if (
+            row_indexer is not None
+            and col_indexer is not None
+            and not isinstance(self._mgr, ArrayManager)
```

Comment:
i think a more general condition would be `self._can_fast_transpose`.  The thing we care about is self.values being cheap.

Reply:
@jorisvandenbossche can you address comments and rebase